runtime/tracing/tracing-core/common/src/aws/smithy/kotlin/runtime/tracing/CoroutineContextUtils.kt

```diff
@@ -45,18 +45,31 @@ public suspend inline fun <R> CoroutineContext.withChildTraceSpan(
     }
 }
 
+@InternalApi
+public class WrappedRootSpan(
```

Comment:
**Question**: Why `@InternalApi`/`public`? Seems like this class is only necessary in this file and could be `private`.

Reply:
it's used in an `inline` function so it can't be private. Options are:

1. Public with internal annotation
2. `internal` with `@PublishedApi`
I have no preference

Reply:
(`@PublishedApi` affects binary compatibility still according to docs)